tests/benchmark_test.py

```diff
@@ -214,3 +214,11 @@ def test_functions_call_error_above(fun):
     with pytest.raises(ValueError):
         fun(above_bounds)
 
+@pytest.mark.parametrize("fun",functions[:3])
+def test_functions_call_shape(fun):
+    """Test all functions give the correct shape"""
+    x_list = numpy.zeros((2,50,fun.dimension))
+    mid_point = numpy.divide(fun.upper_bounds,2)
+    x_list[0,:,:] = numpy.linspace(fun.lower_bounds,mid_point,50)
```

Comment:
I'm unsure about how this works because you are assigning a 1d array into a 2d slice, so I would think that is an error. Here's what happens when I run a similar snippet in my interpreter:
```
>>> import numpy as np
>>> x_list = np.zeros((2, 50, 4))
>>> x_list[0,:,:] = np.linspace(0, 1, 50)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: could not broadcast input array from shape (50,) into shape (50,4)
```
Maybe I am doing something wrong here?

Reply:
The bounds of the function are a list with as many elements as the benchmark function has dimensions. When you input the bounds, you get an array of size `(50, fun.dimension)`, where each column is a linspace for each bound. The glitch arose when trying to pass an array of size `(5, 5000, fun.dimension)` into s266, even though all benchmark function are supposed to be able to handle an input of any size as long as the last axis is the same as `fun.dimension`.

`np.linspace(fun.lower_bounds,fun.upper_bounds)` gets you an array of size `(50, fun.dimension)`, exactly the size needed for `x_list[0,:,:]`

Reply:
Done.